git_machete/cli.py

```diff
@@ -647,8 +661,13 @@ def launch(orig_args: List[str]) -> None:
                 current_branch = git.get_current_branch()
                 machete_client.expect_in_managed_branches(current_branch)
                 machete_client.retarget_github_pr(current_branch)
+            elif github_subcommand == "sync":
+                machete_client.delete_unmanaged(opt_yes=False)
+                machete_client.checkout_github_prs(pr_nos=[],
```

Comment:
Another thing, there's still a surprise checkout within `checkout_github_prs`, even when no managed branch is added:

```
➜ 12:35 ~/git-machete feature/command_clean PR #435 (amalota)$
$ git machete s --color=never 
  master  v3.7.2
  | 
  o-develop  v3.8.0
    | 
    x-chore/nix-update  PR #429  v3.8.0
    | 
    x-chore/deb-install  v3.7.2
    | 
    o-feature/command_clean *  PR #435 (amalota)  v3.8.0

➜ 12:35 ~/git-machete feature/command_clean PR #435 (amalota)$ git machete github sync 
Checking for unmanaged branches...
No branches to delete
Checking for open GitHub PRs...
Checking for untracked managed branches with no downstream...
No branches to delete
➜ 12:35 ~/git-machete chore/nix-update PR #429 $ 
```

So... `checkout_github_prs` switched to my sole PR #429, even though `chore/nix-update` was already managed... I think this behavior is pretty questionable even in `git machete github checkout-prs` (TBD), and surely unexpected in `git machete github sync`/`clean -c`

Reply:
@PawelLipski I am trying to reproduce this but so far i am unable to do so, can you provide exact environment and commands to reproduce it?

Reply:
Hmmm you won't be able to repro with your own account as you've got **more than one** open PR yourself! pls try on a different temporary repo where you have just one PR opened

Reply:
@PawelLipski I remember now, i think we decided that if only 1 pr is being 'checked out' by `checkout_github_prs` then it would get checked out locally at the end, do we not want that anymore? Or do we want to checkout the branch only if it wasn't managed?

Reply:
I would go for the latter

Reply:
Yeah, let's go for the latter: if there's only one branch X matching the given criteria (`--mine`/`--by`/`--all`) AND it has NOT been managed so far &mdash; let's switch the current HEAD to this branch X; otherwise, don't swtich HEAD